test/impl-base/src/main/java/org/jboss/arquillian/test/impl/EventTestRunnerAdaptor.java

```diff
@@ -126,4 +154,26 @@ public void shutdown()
    {
       manager.shutdown();
    }
+   
+   private TestExecutionDecider resolveTestExecutionDecider(Manager manager)
+   {
+       TestExecutionDecider cachedTestExecutionDecider = testExecutionDecider.get();
+
+       if (cachedTestExecutionDecider != null)
+       {
+           return cachedTestExecutionDecider;
+       }
+
+       Validate.notNull(manager, "Manager must be specified.");
+       ServiceLoader serviceLoader = manager.resolve(ServiceLoader.class);
+
+       if (serviceLoader != null)
+       {
+           TestExecutionDecider decider = serviceLoader.onlyOne(TestExecutionDecider.class, DefaultTestExecutionDecider.class);
```

Comment:
@aslakknutsen so lets order these deciders according to precedence and do some chain of it